pkg/workload/workload.go

```diff
@@ -163,9 +167,16 @@ type BatchedTuples struct {
 	// NumTotal is the total number of tuples in all batches. Not all generators
 	// will know this, it's set to 0 when unknown.
 	NumTotal int
-	// Batch is a function to deterministically compute a batch of tuples given
-	// its index.
-	Batch func(int) [][]interface{}
+	// FillBatch is a function to deterministically compute a columnar-batch of
+	// tuples given its index.
+	//
+	// To save allocations, the `Vec`s in the passed `Batch` are reused when
```

Comment:
nit: is it just me or do we not normally enclose types in backticks in comments? obviously take it or leave it